GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1014,13 +1013,7 @@ void LoadUserMenu()
                         DisplayStyle = ToolStripItemDisplayStyle.ImageAndText
                     };
 
-                    button.Click += delegate
-                    {
-                        if (ScriptsRunner.RunScript(script.HotkeyCommandIdentifier, this, RevisionGrid).NeedsGridRefresh)
-                        {
-                            RefreshRevisions();
-                        }
-                    };
+                    button.Click += (s, e) => ScriptsRunner.RunScript(script.HotkeyCommandIdentifier, this, RevisionGrid);
```

Comment:
Previously invoking a script not necessarily resulted in needing to refresh - e.g., running a script `code .` or when the script is run in the background:
https://github.com/gitextensions/gitextensions/blob/04c7dae18856ac65ff1a64ab833344c3d2f9d245/GitUI/ScriptsEngine/ScriptsManager.ScriptRunner.cs#L70-L77
https://github.com/gitextensions/gitextensions/blob/04c7dae18856ac65ff1a64ab833344c3d2f9d245/GitUI/ScriptsEngine/ScriptsManager.ScriptRunner.cs#L143

Now we will always refresh... I don't think this is desirable.

Why aren't we doing this?
```suggestion
                    button.Click += delegate
                    {
                        if (ScriptsRunner.RunScript(script.HotkeyCommandIdentifier, this, RevisionGrid).NeedsGridRefresh)
                        {
                            UICommands.RepoChangedNotifier.Notify();
                        }
                    };```

Reply:
> Now we will always refresh...

No, we do _not_ always refresh.
`UICommands.RepoChangedNotifier.Notify();` is called by the script runner if and only if it returns `NeedsGridRefresh: true`.
This PR should not change the behavior except that it uses the "official" notification (which might inform additional minor subscribers, which is desirable). 

> Why aren't we doing this?

Why should all callers need to do this? The repo was changed by a script, so the _script runner_ should notify _every_ subscriber.

Reply:
> The repo was changed by a script, so the _script runner_ should notify _every_ subscriber.

That's the thing we don't know that, and it's wrong to assume that the script changed the state. I've given you an example above. I have scripts that just opne VS or VS Code in the repo, or do some other aux tasks that _do not_ change the state of the repo that warrant a refresh.
Refreshing can be an **_extremely_** expensive operation - e.g., it takes several minutes to refresh the VS repo because there 100K tags in it.